Gemfile

```diff
@@ -0,0 +1,62 @@
+source 'https://rubygems.org'
+git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+
+ruby '3.1.2'
+
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails', branch: 'main'
+gem 'rails', '~> 6.1.6', '>= 6.1.6.1'
+# Use sqlite3 as the database for Active Record
+gem 'sqlite3', '~> 1.4'
```

Comment:
e đọc lưu ý ứng vs từng chương nhé

Reply:
https://docs.google.com/spreadsheets/d/1ME3lk9-kYoQwC-3Exrq9ZkA8F6o0ZJxgGH8PKru7Bug/edit#gid=1196249713

Reply:
thay thế gem sqlite3 bằng gem mysql2

Reply:
thay thế toàn bộ nháy ' thành nháy "